### ISSUE TITLE
Will this work with Private Repos like Nexus ? 

### ISSUE THREAD

**Issue by mikekuzak** [linuxserver/fleet#10]
I did not see any clear indication in the docs if this is able to work with private repos like Nexus OSS or Artifactory, Apache Archiva.

**Comment by JoshStark** [comment]
Hi,

No it is currently not compliant with private repositories, although future work will introduce this functionality.

**Comment by mikekuzak** [comment]
@JoshStark I can help codding this. Not sure if you have already a design, direction how to approach this. I was planning to start with Nexus which has a well defined API. Haven't had time to look at the current classes, but saw a Interface for Service Provider.

**Comment by JoshStark** [comment]
@mikekuzak Yeah, there is an interface `DockerApiClient` which is the basis for interactions with DockerHub at present. It's used in `DockerApiDelegate` which is in turn used by various schedules and actions. The sole implementation so far is `DockerHubApiClient` but I'll also need to create a `GitHubApiClient` to fit with LSIO's immediate needs. If you fancy having a go with Nexus, feel free to fork and PR. I can have a look after that!
